homeassistant/components/wmspro/config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Config flow for WMS WebControl pro API integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import aiohttp
+import voluptuous as vol
+from wmspro.webcontrol import WebControlPro
+
+from homeassistant.components import dhcp
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN, SUGGESTED_HOST
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST, description={"suggested_value": SUGGESTED_HOST}): str,
+    }
+)
+
+
+async def setup_and_verify(hass: HomeAssistant, host: str) -> WebControlPro:
+    """Set up and verify the connection to the wmspro."""
+    session = async_get_clientsession(hass)
+
+    hub = WebControlPro(host, session)
+    try:
+        pong = await hub.ping()
+    except aiohttp.ClientError as err:
+        raise CannotConnect from err
+    if not pong:
+        raise CannotConnect
+
+    return hub
+
+
+class WebControlProConfigFlow(ConfigFlow, domain=DOMAIN):
```

Comment:
Do we have something unique we can set as unique identifier?

Reply:
Unfortunately not, the hub API doesn't return anything, no serial or mac of the hub itself.

Reply:
Can we then add `self._async_abort_entries_match` after the `if user_input is not None` where we guard if the same host is already configured?